### PR TITLE
Optional TLS/Authentication for Contact Support Menu

### PR DESCRIPTION
**Pull request by dmccoystephenson**
## Problem
In the event that an email server does not support TLS or authentication, the Contact Support Menu ceases to function.

## Solution
TLS and authentication have been made optional via the introduction of the CSM_TLS_ENABLED and CSM_AUTH_ENABLED environment variables.

## Testing
- Unit tests have been added and verified to be passing.
- Sending an email through Google's email server has been verified to work with TLS / AUTH enabled.
- Sending an email to a locally-running mail server has been verified to work with TLS / AUTH disabled.
- Sending an email through WYDOT's mail server has been verified to work with TLS / AUTH disabled.

### Note
It should be noted that if pretty formatting is enabled, WYDOT's email server won't accept & forward the email. Pretty formatting is explicitly disabled in the contact support menu due to this.

### CODE REVIEW

File: services/api/src/emailSender.py [hunk 1]
```diff
@@ -9,31 +12,45 @@ def __init__(self, smtp_server, port):
...
-
             # prepare email
-            toSend = self.prepareEmailToSend(
+            toSend = ""
+            toSend = self.format(
                 sender, recipient, subject, message, replyEmail
             )
+            if tlsEnabled == "true":
+                self.server.starttls(context=self.context)  # start TLS encryption
+                self.server.ehlo()  # say hello
```
Comment (payneBrandon) [inline services/api/src/emailSender.py]: Do we not need to ehlo for non-tls systems?

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just the one question, otherwise looks good!

File: services/api/src/emailSender.py (hunk 1 above)
Comment (dmccoystephenson) [reply services/api/src/emailSender.py]: Apparently this function is implicitly called by .sendmail() if needed, so we do not need to explicitly call it here.

Source: https://realpython.com/python-send-email/#option-2-using-starttls

**Comment by payneBrandon** [comment]
@dmccoystephenson looks like a couple of merge conflicts

**Comment by dmccoystephenson** [comment]
Keeping this as a draft for now until I can re-verify the functionality for this PR

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Pull request merged**